### DISCUSSION TITLE
use of function zen_draw_textarea_field: cols parameter may be % or missing

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5815] [Ideas]
The cols parameter should be an integer, if used.
![Clipboardmeta](https://user-images.githubusercontent.com/4391026/99196888-5e402200-278f-11eb-9dc9-0cde5c880d37.gif)

Throughout ZC the use of the col parameter in the function call may be integer, a number as a string, or a percentage (invalid), or even empty, producing a col in the html output with no value.

The function assumes the col/width and row/height parameters are always supplied, and are used "as is" with no casting.

Given there are almost 60 instances of this function call, maybe it would be simplest to adjust the function to deal with an empty cols/width parameter and cast the inputs?

**Comment by drbyte** [comment]
I don't see any references to `%` in calls to `zen_draw_textarea_field` in v158a.

There are a few places where an empty `cols` value is passed to the function, but it is cast to `(int)` and only rendered if `> 0`.
